packages/@uppy/companion/test/__tests__/uploader.js

```diff
@@ -183,14 +185,27 @@ describe('uploader with tus protocol', () => {
   }
 
   test('upload functions with xhr protocol', async () => {
-    nock('http://localhost').post('/').reply(200)
-
-    const ret = await runMultipartTest()
-    expect(ret).toMatchObject({ url: null, extraData: { response: expect.anything(), bytesUploaded: 17 } })
+    let alreadyCalled = false
+    const server = createServer((req,res) => {
```

Comment:
```suggestion
    // We are creating our own test server for this test
    // instead of nock because got 14 only supports Node.js >20.x
    // and streams are handled differently in Node.js before that and in nock.
    // Ref: https://github.com/nock/nock/issues/2595
    const server = createServer((req,res) => {
```